advisor/src/main/kotlin/advisors/Osv.kt

```diff
@@ -185,13 +185,12 @@ private fun Vulnerability.toOrtVulnerability(): org.ossreviewtoolkit.model.Vulne
         }
     } ?: (null to null)
 
-    if (severity == null && databaseSpecific != null) {
+    val specificSeverity = databaseSpecific?.get("severity")
+    if (severity == null && specificSeverity != null) {
         // Fallback to the 'severity' property of the unspecified 'databaseSpecific' object.
         // See also https://github.com/google/osv.dev/issues/484.
-        databaseSpecific!!["severity"]?.let {
-            if (it is JsonPrimitive) {
-                severity = it.contentOrNull
-            }
+        if (specificSeverity is JsonPrimitive) {
```

Comment:
I don't see any problem with the assertion as was before, do you?
I wonder if the detect rule flags all assertions and if we should disable that rule.
Mis-use IMO can be spotted in code reviews.

What do you think?

Reply:
> I don't see any problem with the assertion as was before, do you?

The use of `!!` per se is considered bad practice (by Detekt and [common](https://unity-technologies.github.io/kotlin-guide/#safe-variable-declarations) [Kotlin](https://medium.com/android-news/how-to-remove-all-from-your-kotlin-code-87dc2c9767fb) [guidelines](https://www.baeldung.com/kotlin/not-null-assertion#use-cases)), independently of the context, and personally agree to that. IMO we should avoid the use of `!!` where ever possible, for readability and safety. To me, `!!` is like an unchecked low-level cast in C++; you wouldn't do that either, unless absolutely necessary.

> I wonder if the detect rule flags all assertions and if we should disable that rule.

Yes, Detekt (with type resolution enabled) flags the use of `!!` in general, and I believe that rule to be good, so I wouldn't disable it.

> Mis-use IMO can be spotted in code reviews.

I believe a tiny `!!` can easily be overlooked, and differentiating a misuse from proper use can be challenging without a very deep review, which we might not always have time to do.

Reply:
Ok, let's keep the code as you wrote it then.